lib/include/openamp/virtio.h

```diff
@@ -227,6 +227,12 @@ __deprecated void virtio_describe(struct virtio_device *dev, const char *msg,
  */
 
 struct virtio_dispatch {
+	/** Alloc buffer from transport layer heap */
+	void *(*alloc_buf)(struct virtio_device *vdev, size_t size, size_t align);
+
+	/** Free buffer to transport layer heap */
+	void (*free_buf)(struct virtio_device *vdev, void *buf);
+
```

Comment:
Seems to me that it is not the good place to add these functions
The memory allocation/free is independent from the virtio layer as addressed directly to the transport layer.
The `virtio_dispatch` is used for communication between the virtio layer and the transport layer
What about  a new structure similar to  `virtio_dispatch` ?  `virtio_mem_ops` or a better name...

Reply:
Seems better, I will try this, Thanks.

Reply:
Done, please review again @arnopo